test/store_unittest.cc

```diff
@@ -0,0 +1,226 @@
+#include "../src/configuration.h"
+#include "../src/resource.h"
+#include "../src/store.h"
+#include "../src/time.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+class MetadataStoreTest : public ::testing::Test {
+ protected:
+  Configuration config;
+  MetadataStore store;
+
+  MetadataStoreTest() : config(), store(config) {}
+
+  std::map<MonitoredResource, MetadataStore::Metadata> GetMetadataMap() {
+    return store.GetMetadataMap();
+  }
+
+  void PurgeDeletedEntries() {
+    store.PurgeDeletedEntries();
+  }
+
+};
+
+TEST_F(MetadataStoreTest, ResourceID) {
+  MonitoredResource mr("some_resource", {});
+  store.UpdateResource({"id"}, mr);
+  EXPECT_EQ("some_resource", store.LookupResource("id").type());
+}
+
+TEST_F(MetadataStoreTest, ResourceNotFound) {
+  EXPECT_THROW(store.LookupResource("some_resource_id"), std::out_of_range);
+}
+
+TEST_F(MetadataStoreTest, ManyResources) {
+  MonitoredResource mr1("some_resource1", {});
+  MonitoredResource mr2("some_resource2", {});
+  store.UpdateResource({"id1"}, mr1);
+  store.UpdateResource({"id2"}, mr2);
+  EXPECT_EQ("some_resource1", store.LookupResource("id1").type());
+  EXPECT_EQ("some_resource2", store.LookupResource("id2").type());
+}
+
+TEST_F(MetadataStoreTest, ManyResourceIDs) {
+  MonitoredResource mr("some_resource", {});
+  store.UpdateResource({"id1", "id2", "id3"}, mr);
+  EXPECT_EQ("some_resource", store.LookupResource("id1").type());
+  EXPECT_EQ("some_resource", store.LookupResource("id2").type());
+  EXPECT_EQ("some_resource", store.LookupResource("id3").type());
+}
+
+TEST_F(MetadataStoreTest, ManyIdsAndResources) {
+  MonitoredResource mr1("some_resource1", {});
+  MonitoredResource mr2("some_resource2", {});
+  store.UpdateResource({"id1", "id2", "id3"}, mr1);
+  store.UpdateResource({"id-a", "id-b", "id-c"}, mr2);
+  EXPECT_EQ("some_resource1", store.LookupResource("id1").type());
+  EXPECT_EQ("some_resource1", store.LookupResource("id2").type());
+  EXPECT_EQ("some_resource1", store.LookupResource("id3").type());
+  EXPECT_EQ("some_resource2", store.LookupResource("id-a").type());
+  EXPECT_EQ("some_resource2", store.LookupResource("id-b").type());
+  EXPECT_EQ("some_resource2", store.LookupResource("id-c").type());
+}
+
+TEST_F(MetadataStoreTest, MultipleResourceIdsUpdate) {
+  MonitoredResource mr("some_resource", {});
+  store.UpdateResource({"id1", "id2"}, mr);
+  EXPECT_EQ("some_resource", store.LookupResource("id1").type());
+  EXPECT_EQ("some_resource", store.LookupResource("id2").type());
+  store.UpdateResource({"id-a", "id-b"}, mr);
+  EXPECT_EQ("some_resource", store.LookupResource("id-a").type());
+  EXPECT_EQ("some_resource", store.LookupResource("id-b").type());
```

Comment:
We should check whether `"id1"` and `"id2"` still map to the resource. This will document the current behavior, and allow a discussion of whether that's the behavior we want.

Reply:
Done.